Application/Features/Queries/ShowAvailableReservations/ShowAvailableReservationsQueryHandler.cs

```diff
@@ -36,15 +36,18 @@ public class ShowAvailableReservationsQueryHandler : IRequestHandler<ShowAvailab
 
         var availableReservations = new List<AvailableReservationDto>();
 
-        foreach (var reservation in reservations)
+        foreach (var movie in movies)
         {
-            var movie = movies.FirstOrDefault(x => x.Id == reservation.MovieId);
-            if (movie is not null)
+            var reservationList = reservations.FindAll(x => x.MovieId == movie.Id);
```

Comment:
Ovde se i dalje radi sa listom svih rezervacija. Treba kao ono što smo rekli - za svaki MovieId zoveš reservation API pojedinačno